src/System.Private.ServiceModel/tools/test/SelfHostWcfService/WcfService.cs

```diff
@@ -13,9 +13,9 @@ namespace WcfService
 {
     internal class WcfService : IWcfService
     {
-        public String EchoWithTimeout(String message)
+        public String EchoWithTimeout(String message, TimeSpan timeout)
```

Comment:
If there's a concern about having a thread blocking for the timeout amount of time, change the code (and interface) to this:
```csharp
        public async Task<String> EchoWithTimeout(String message, TimeSpan timeout)
        {
            await Task.Delay(timeout);
            return message;
        }